string packageName
string launchFileName
---
string launchFilePath
